From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 6 Apr 2020 12:05:58 +0200
Subject: wayland: Always update unconfigured size

If the window size didn't change, we should still update the
unconfigured size as the window size might not be up to date.

Origin: upstream, 3.24.18
---
 gdk/wayland/gdkwindow-wayland.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index 34ac430..8e2e023 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -1145,10 +1145,6 @@ gdk_wayland_window_maybe_configure (GdkWindow *window,
   gboolean is_xdg_popup;
   gboolean is_visible;
 
-  if (window->width == width &&
-      window->height == height &&
-      impl->scale == scale)
-    return;
 
   if (needs_initial_configure (window) &&
       !impl->initial_configure_received)
@@ -1158,6 +1154,11 @@ gdk_wayland_window_maybe_configure (GdkWindow *window,
       return;
     }
 
+  if (window->width == width &&
+      window->height == height &&
+      impl->scale == scale)
+    return;
+
   /* For xdg_popup using an xdg_positioner, there is a race condition if
    * the application tries to change the size after it's mapped, but before
    * the initial configure is received, so hide and show the surface again
